### PR TITLE
added option to also re-insert

### PR DESCRIPTION
**Pull request by gluneau**
It deletes, the re-inserts

By calling the script with the insertminc argument, the minc will be inserted as well.

`tools/deletemincsqlwrapper.pl -profile prod -insertminc`

### CODE REVIEW

File: tools/deletemincsqlwrapper.pl [hunk 1]
```diff
@@ -8,27 +8,36 @@
 use Term::ANSIColor qw(:constants);
 use NeuroDB::DBI;
 
-my $profile = "prod";
+my $profile = "null_grads"; # "prod";
```
Comment (MounaSafiHarab) [inline tools/deletemincsqlwrapper.pl]: should not be hard coded; so maybe separate into 2 pull requests

File: tools/deletemincsqlwrapper.pl [hunk 2]
```diff
@@ -12,23 +12,32 @@
...
+     or (pf.ParameterTypeID=333 and pf.`VALUE` like '%0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.%')
+     or (pf.ParameterTypeID=334 and pf.`VALUE` like '%0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.%')
+     or (pf.ParameterTypeID=343 and pf.`VALUE` like '%0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.%'))
+    ORDER BY f.FileID
```
Comment (MounaSafiHarab) [inline tools/deletemincsqlwrapper.pl]: @gluneau 

all these hard coded parameter TypeID won't work for other projects; same for SeriesUID

**Comment by MounaSafiHarab** [comment]
@gluneau 

1) do you think a script description/usage would be helpful? since you are adding an argument, it is not very clear how to run the script

2) since there is an option to insert deleted files, I am not sure the filename `deletemincsqlwrapper.pl`  is still accurate.

**Comment by gluneau** [comment]
@MounaSafiHarab Please check the requested changes.

**Comment by MounaSafiHarab** [comment]
Mouna to test it and eventually make this more generic for all projects to be able to use (query is IBIS specific now)

**Comment by MounaSafiHarab** [comment]
@gluneau 

I tested it and I ran into a couple of issues:
1) I had to update the help section and added the `use Getopt::Tabular` and `my @opt_table` sections to reflect the usage of the `-insertminc` option (you can then make the `prod` file name an argument as well so it is not hard coded)
2) More importantly, the minc_deletion, when it is called, is not given the tarchiveLocation as an argument, so tarchiveLoader does not run.

File: tools/deletemincsqlwrapper.pl [hunk 3]
```diff
@@ -69,5 +76,11 @@
     my $minc_delete_log = `$minc_delete_cmd`;
     print $minc_delete_log . "\n";
 
+    if ($insertminc) {
+      my $tar_loader_cmd  = "../uploadNeuroDB/tarchiveLoader -profile " . $profile . " -seriesuid " . $fF->{'SeriesUID'} . " -verbose -globLocation " . $fF->{'ArchiveLocation'};
+      print $tar_loader_cmd . "\n";
```
Comment (MounaSafiHarab) [inline tools/deletemincsqlwrapper.pl]: @gluneau 
not sure where the ArchiveLocation is coming from. You need to join on the tarchive table to get this location.
(running this with the `insertminc` option gives the error:
```../uploadNeuroDB/tarchiveLoader -profile prod -seriesuid 1.3.12.2.1107.5.2.43.67007.30000016111018154618200000794 -verbose -globLocation 
...
usage: ../uploadNeuroDB/tarchiveLoader </path/to/DICOM-tarchive> [options]
       ../uploadNeuroDB/tarchiveLoader -help to list options


	ERROR: You must specify a valid tarchive and an existing profile.
```

**Comment by MounaSafiHarab** [comment]
closed but same changes were merged under this pull request:
https://github.com/aces/Loris-MRI/pull/193

**Pull request closed without merging**